.github/workflows/container.yml

```
@@ -195,24 +195,48 @@ jobs:
         run: podman images | grep 'ghcr.io/parca-dev/parca-agent'
 
       - name: Install cosign
-        if: ${{ github.event_name != 'pull_request' }}
         uses: sigstore/cosign-installer@48866aa521d8bf870604709cd43ec2f602d03ff2 # tag=v2.4.1
 
       - name: Login to registry
-        if: ${{ github.event_name != 'pull_request' }}
         run: |
           echo "${{ secrets.GITHUB_TOKEN }}" | podman login -u parca-dev --password-stdin ghcr.io
           echo "${{ secrets.QUAY_PASSWORD }}" | cosign login -u "${{ secrets.QUAY_USERNAME }}" --password-stdin quay.io
 
       - name: Install crane
-        if: ${{ github.event_name != 'pull_request' }}
         uses: imjasonh/setup-crane@5146f708a817ea23476677995bf2133943b9be0b # tag=v0.1
 
       - name: Push and sign container
-        if: ${{ github.event_name != 'pull_request' }}
         env:
           COSIGN_EXPERIMENTAL: true
         run: |
-          make push-container
-          make sign-container
-          make push-signed-quay-container
+          make push-container-e2e
+          if [[ ${{ github.event_name}} != 'pull_request' ]]; then
+              make push-container
+              make sign-container
+              make push-signed-quay-container
+          fi
+
+  run-smoke-tests:
```

```suggestion
  run-end-to-end-tests:
```
nit: e2e or smoke? Let's be consistent.

---

Good catch!